frontend/src/app/oe/components/base-box-hor/base-box-hor.component.ts

```diff
@@ -39,20 +41,24 @@ export class BaseBoxHorComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
   }
 
+  navigateTo(): void {
+    !isTextSelection() && this.router.navigate([this.link]);
+  }
+
   toHHMMSS(secs) {
```

Comment:
I see basebox.component.ts contains the same function. I suggest to refactor it into reusable one.